framework/src/modules/chain/bft/bft.js

```diff
@@ -112,14 +113,15 @@ class BFT extends EventEmitter {
 			aboveHeight: removeFromHeight - 1,
 		});
 
-		// Make sure there are 2 rounds of block headers available
+		// Make sure there are BFT_ROUND_THRESHOLD rounds of block headers available
 		if (
 			this.finalityManager.maxHeight - this.finalityManager.minHeight <
-			this.constants.activeDelegates * 2
+			this.constants.activeDelegates * BFT_ROUND_THRESHOLD
 		) {
 			const tillHeight = this.finalityManager.minHeight - 1;
 			const fromHeight =
-				this.finalityManager.maxHeight - this.constants.activeDelegates * 2;
+				this.finalityManager.maxHeight -
+				this.constants.activeDelegates * BFT_ROUND_THRESHOLD;
```

Comment:
The LIP defined 

> we need to ensure that the block headers of the blocks at heights A.height-302 up to height A.height are contained in memory

https://github.com/LiskHQ/lips/blob/master/proposals/lip-0014.md#reverting-blocks

This becomes having exactly 303 blocks in headers. Above logic will lead to 304 Blocks instead. Remember this `tillHeight` and `fromHeight` are inclusive. If we already have headers from `451-590`, above calculation will make sure to have `287-590`, this range when inclusive become 304 blocks.